inc/class-interest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Interest functionality.
+ *
+ * @package Pantheon/EdgeIntegrations
+ */
+
+namespace Pantheon\EI\WP;
+
+use Pantheon\EI\WP\Infrastructure\Service;
+
+/**
+ * Interests class.
+ */
+final class Interest extends Service {
+
+	/**
+	 * Performs setup operations.
+	 */
+	public function register() {
+		add_action( 'wp_enqueue_scripts', [ $this, 'register_script' ] );
+		add_action( 'the_post', [ $this, 'localize_interests' ] );
+	}
+
+	/**
+	 * Registers the script.
+	 *
+	 * @return void
+	 */
+	public function register_script() {
+		wp_enqueue_script( 'pantheon-ei-interest', plugins_url( '/dist/js/assets.js', PANTHEON_EDGE_INTEGRATIONS_FILE ), [], PANTHEON_EDGE_INTEGRATIONS_VERSION, true );
+	}
+
+	/**
+	 * Localizes the script.
+	 *
+	 * @param WP_Post $post_object The current post.
+	 */
+	public function localize_interests( $post_object ) {
+		/**
+		 * Allow engineers to modify post type support.
+		 *
+		 * @hook pantheon.ei.post_types
+		 * @param {array} Post types to target for interests.
```

Comment:
Nit: PHP docblock syntax is slightly different than JSDoc syntax. This should be:

```suggestion
		 * @param array $post_object Post types to target for interests.
```

🔢

Reply:
:doh: went from JS > PHP. Addressed in [8bcf560](https://github.com/pantheon-systems/pantheon-wordpress-edge-integrations/pull/8/commits/8bcf5609a0ee9de8890b1f70cdc01661a91aa88e).